airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence2.java

```diff
@@ -0,0 +1,1911 @@
+import static io.airbyte.db.instance.configs.jooq.Tables.CONNECTION;
+import static io.airbyte.db.instance.configs.jooq.Tables.CONNECTION_OPERATION;
+import static io.airbyte.db.instance.configs.jooq.Tables.OPERATION;
+import static io.airbyte.db.instance.configs.jooq.Tables.STATE;
+import static io.airbyte.db.instance.configs.jooq.Tables.WORKSPACE;
+import static org.jooq.impl.DSL.asterisk;
+import static org.jooq.impl.DSL.select;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Sets;
+import io.airbyte.commons.enums.Enums;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.util.MoreIterators;
+import io.airbyte.commons.version.AirbyteVersion;
+import io.airbyte.config.AirbyteConfig;
+import io.airbyte.config.ConfigSchema;
+import io.airbyte.config.ConfigWithMetadata;
+import io.airbyte.config.Configs;
+import io.airbyte.config.DestinationConnection;
+import io.airbyte.config.DestinationOAuthParameter;
+import io.airbyte.config.JobSyncConfig.NamespaceDefinitionType;
+import io.airbyte.config.Notification;
+import io.airbyte.config.OperatorDbt;
+import io.airbyte.config.OperatorNormalization;
+import io.airbyte.config.ResourceRequirements;
+import io.airbyte.config.Schedule;
+import io.airbyte.config.SourceConnection;
+import io.airbyte.config.SourceOAuthParameter;
+import io.airbyte.config.StandardDestinationDefinition;
+import io.airbyte.config.StandardSourceDefinition;
+import io.airbyte.config.StandardSourceDefinition.SourceType;
+import io.airbyte.config.StandardSync;
+import io.airbyte.config.StandardSync.Status;
+import io.airbyte.config.StandardSyncOperation;
+import io.airbyte.config.StandardSyncOperation.OperatorType;
+import io.airbyte.config.StandardSyncState;
+import io.airbyte.config.StandardWorkspace;
+import io.airbyte.config.State;
+import io.airbyte.db.Database;
+import io.airbyte.db.ExceptionWrappingDatabase;
+import io.airbyte.db.instance.configs.jooq.enums.ActorType;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.ConnectorSpecification;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.jooq.DSLContext;
+import org.jooq.JSONB;
+import org.jooq.Record;
+import org.jooq.Record1;
+import org.jooq.Result;
+import org.jooq.impl.DSL;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DatabaseConfigPersistence2 implements ConfigPersistence {
```

Comment:
This large class makes me think it's probably times for each database model to have it's own persistence class instead of having a single class responsible for all this. We don't have to do this now to avoid changing too many things at once, but I'm thinking we should inset a comment here that says we should break away from this pattern moving forward and slowly migrate the older tables over. 

How does this sound? @cgardens ?

Reply:
@davinchia as discussed over call, thats the plan. For now we are just changing the DatabaseConfigPersistence so that we dont have to change any application level code. As part of leveraging the new table structure we would head into that direction